# **ESTUDO DE CASO: ANÁLISE EXPLORATÓRIA DE DADOS**
#### **Case**: *Papelaria*
#### **Descrição**: *A área de analytics de uma empresa de varejo em papelaria precisa responder uma série de questões de negócio relacionadas às vendas do último mês. Os resultados ajudarão a empresa a priorizar o estoque e reposição dos itens mais vendidos (ou mais rentáveis); otimizar o mix de produtos oferecidos na loja; e tomar decisões de precificação e negociação com fornecedores.*
#### **Lista de Variáveis**:
- ##### *COD_PRODUTO: Código de identificação do produto*
- ##### *CATEGORIA_PRODUTO: Categoria do produto*
- ##### *SECAO_PRODUTO: Seção/departamento do produto*
- ##### *PRECO_UNITARIO: Preço unitário de venda do produto, vigente no mês anterior*
- ##### *QTD_UNIDADES_VENDIDAS_MES: Quantidade de unidades vendidas do produto no último mês*

#### **Base de Dados**: *Papelaria.txt*
--------------

#### **Tarefa**: Responda as 8 questões a seguir, utilizando as técnicas de análise exploratória de dados e visualização de dados mais pertinentes em cada situação. Para todas as questões, descreva os resultados obtidos e escreva suas conclusões. Cada questão vale 1,25 pontos.

#### **Entregável**: Este mesmo notebook do Python (extensão .ipynb) já compilado, com seus códigos em Python e conclusões. Não altere a estrutura geral do notebook.

#### **Data máxima de entrega**: 24/06/2025

------------


### **CARREGAMENTO DE BIBLIOTECAS**

In [1]:
# Códigos...
import pandas as pd
import numpy as np

### **LEITURA DA BASE DE DADOS**

In [ ]:
# Códigos...
dados = pd.read_csv("D:\TELMA\Documents\Git\MBA_IA\Analise_Exploratoria_de_Dados\Papelaria.txt", sep='\t', header=0)

In [3]:
dados.head()

,COD_PRODUTO,CATEGORIA_PRODUTO,SECAO_PRODUTO,PRECO_UNITARIO,QTD_UNIDADES_VENDIDAS_MES
0,COD_P_233209,Tinta e Pincel,Papelaria,5.9,14
1,COD_P_235103,Caneta Especial,Papelaria,5.9,26
2,COD_P_178413,Monitor,Informática,799.9,3
3,COD_P_316368,Mochila Escolar,Acessórios,139.0,3
4,COD_P_190632,Suporte de Celular,Acessórios,29.9,5


### **QUESTÃO 1**

#### Quais categorias de produto possuem produtos maior preço unitário, em média?

In [4]:
# Insira seus códigos em Python aqui (divida em várias células, se desejar)
cat_produtos = dados.groupby('CATEGORIA_PRODUTO')['PRECO_UNITARIO'].mean().sort_values(ascending=False)

In [5]:
cat_produtos.head()

CATEGORIA_PRODUTO
Notebook Premium      5216.633333
Notebook              3779.900000
Tablet                2919.700000
Impressora a Laser    1479.950000
Monitor                887.150000
Name: PRECO_UNITARIO, dtype: float64

Insira a descrição dos resultados / conclusões aqui

Verificamos que as categorias de produto com maior preço unitário são produtos de informática. Os cinco primeiros colocados são:

1) Notebook Premium
2) Notebook
3) Tablet
4) Impressora a laser
5) Monitor

### **QUESTÃO 2**

#### Dentro de cada seção, quais categorias de produtos mais contribuíram com o faturamento bruto (produto entre a quantidade de unidades vendidas e o preço unitário)?

In [6]:
# Insira seus códigos em Python aqui (divida em várias células, se desejar)
dados['FATURAMENTO_BRUTO'] = dados['QTD_UNIDADES_VENDIDAS_MES'] * dados['PRECO_UNITARIO']
dados.head(10)

,COD_PRODUTO,CATEGORIA_PRODUTO,SECAO_PRODUTO,PRECO_UNITARIO,QTD_UNIDADES_VENDIDAS_MES,FATURAMENTO_BRUTO
0,COD_P_233209,Tinta e Pincel,Papelaria,5.9,14,82.6
1,COD_P_235103,Caneta Especial,Papelaria,5.9,26,153.4
2,COD_P_178413,Monitor,Informática,799.9,3,2399.7
3,COD_P_316368,Mochila Escolar,Acessórios,139.0,3,417.0
4,COD_P_190632,Suporte de Celular,Acessórios,29.9,5,149.5
5,COD_P_168517,Tinta e Pincel,Papelaria,7.9,6,47.4
6,COD_P_220491,Papéis e Cartolina,Papelaria,14.9,6,89.4
7,COD_P_310383,Impressora a Jato,Informática,399.9,7,2799.3
8,COD_P_231329,Cartucho de Tinta,Informática,139.9,12,1678.8
9,COD_P_309533,Grampeador e Perfurador,Papelaria,28.9,2,57.8


In [7]:

dados_1 = dados.groupby(["SECAO_PRODUTO", "CATEGORIA_PRODUTO"], as_index=False) \
                .agg(FATURAMENTO_BRUTO_TOTAL=("FATURAMENTO_BRUTO", "sum")) \
                .sort_values(by=["SECAO_PRODUTO", "FATURAMENTO_BRUTO_TOTAL"], ascending=[True, False])

In [8]:
dados_1.head(10)

,SECAO_PRODUTO,CATEGORIA_PRODUTO,FATURAMENTO_BRUTO_TOTAL
7,Acessórios,Mochila Escolar,6203.2
6,Acessórios,Fone de Ouvido sem Fio,4274.0
5,Acessórios,Fone de Ouvido com Fio,4076.0
8,Acessórios,Mochila Executiva,3058.9
1,Acessórios,Caixa de Som Bluetooth,2118.6
4,Acessórios,Carregador Portátil,1645.6
10,Acessórios,Suporte de Monitor,1353.9
0,Acessórios,Cabo USB,822.8
3,Acessórios,Calculadora Simples,668.0
9,Acessórios,Suporte de Celular,550.6


In [9]:
dados_1["rank"] = dados_1.groupby("SECAO_PRODUTO")["FATURAMENTO_BRUTO_TOTAL"].rank(method="first", ascending=False)

dados_1.head(100)

,SECAO_PRODUTO,CATEGORIA_PRODUTO,FATURAMENTO_BRUTO_TOTAL,rank
7,Acessórios,Mochila Escolar,6203.20,1.0
6,Acessórios,Fone de Ouvido sem Fio,4274.00,2.0
5,Acessórios,Fone de Ouvido com Fio,4076.00,3.0
8,Acessórios,Mochila Executiva,3058.90,4.0
1,Acessórios,Caixa de Som Bluetooth,2118.60,5.0
4,Acessórios,Carregador Portátil,1645.60,6.0
10,Acessórios,Suporte de Monitor,1353.90,7.0
0,Acessórios,Cabo USB,822.80,8.0
3,Acessórios,Calculadora Simples,668.00,9.0
9,Acessórios,Suporte de Celular,550.60,10.0


In [10]:
dados_2 = dados_1.sort_values(by=["SECAO_PRODUTO", "FATURAMENTO_BRUTO_TOTAL"], ascending=[True, False])


In [11]:
# Filtrar linhas com rank < 5 e exibir apenas as colunas desejadas
filtro = dados_2[dados_2["rank"] <= 5][["SECAO_PRODUTO", "CATEGORIA_PRODUTO", "FATURAMENTO_BRUTO_TOTAL", "rank"]]
filtro

,SECAO_PRODUTO,CATEGORIA_PRODUTO,FATURAMENTO_BRUTO_TOTAL,rank
7,Acessórios,Mochila Escolar,6203.2,1.0
6,Acessórios,Fone de Ouvido sem Fio,4274.0,2.0
5,Acessórios,Fone de Ouvido com Fio,4076.0,3.0
8,Acessórios,Mochila Executiva,3058.9,4.0
1,Acessórios,Caixa de Som Bluetooth,2118.6,5.0
22,Informática,Tablet,50775.3,1.0
19,Informática,Notebook,20999.4,2.0
16,Informática,Monitor,16444.8,3.0
12,Informática,Cartucho de Tinta,11671.0,4.0
14,Informática,Impressora a Jato,11507.8,5.0


Insira a descrição dos resultados / conclusões aqui

Podemos verificar na célula acima as 5 categorias com maior faturamento de cada seção. São eles:

Seção Acessórios:
    1) Mochila escolar
    2) Fone de Ouvido sem Fio
    3) Fone de Ouvido com Fio
    4) Mochila Executiva
    5) Caixa de Som Bluetooth

Seção Informática:
    1) Tablet
    2) Notebook
    3) Monitor
    4) Cartucho de Tinta
    5) Impressora a Jato

Seção Papelaria:
    1) Lápis de Cor e Giz
    2) Caneta Especial
    3) Agenda
    4) Caderno 10 Matérias	
    5) Pasta

### **QUESTÃO 3**

#### Existe relação entre o preço unitário e a quantidade de unidades vendidas? Essa relação muda a depender da seção do produto?

In [12]:
dados.head(20)

,COD_PRODUTO,CATEGORIA_PRODUTO,SECAO_PRODUTO,PRECO_UNITARIO,QTD_UNIDADES_VENDIDAS_MES,FATURAMENTO_BRUTO
0,COD_P_233209,Tinta e Pincel,Papelaria,5.9,14,82.6
1,COD_P_235103,Caneta Especial,Papelaria,5.9,26,153.4
2,COD_P_178413,Monitor,Informática,799.9,3,2399.7
3,COD_P_316368,Mochila Escolar,Acessórios,139.0,3,417.0
4,COD_P_190632,Suporte de Celular,Acessórios,29.9,5,149.5
5,COD_P_168517,Tinta e Pincel,Papelaria,7.9,6,47.4
6,COD_P_220491,Papéis e Cartolina,Papelaria,14.9,6,89.4
7,COD_P_310383,Impressora a Jato,Informática,399.9,7,2799.3
8,COD_P_231329,Cartucho de Tinta,Informática,139.9,12,1678.8
9,COD_P_309533,Grampeador e Perfurador,Papelaria,28.9,2,57.8


In [13]:
# Insira seus códigos em Python aqui (divida em várias células, se desejar)
dados_q3 = dados.groupby(["SECAO_PRODUTO", "CATEGORIA_PRODUTO"], as_index=False) \
                .agg(FATURAMENTO_BRUTO_TOTAL=("FATURAMENTO_BRUTO", "sum"), 
                     QTD_UNIDADES_VENDIDAS_MES_TOTAL=("QTD_UNIDADES_VENDIDAS_MES", "sum"),
                     PRECO_UNITARIO_MEDIO=("PRECO_UNITARIO", "mean")
                     )\
                .sort_values(by=["SECAO_PRODUTO", "QTD_UNIDADES_VENDIDAS_MES_TOTAL"], ascending=[True, False])


In [14]:
dados_q3[dados_q3["SECAO_PRODUTO"] == "Acessórios"].head(20)

,SECAO_PRODUTO,CATEGORIA_PRODUTO,FATURAMENTO_BRUTO_TOTAL,QTD_UNIDADES_VENDIDAS_MES_TOTAL,PRECO_UNITARIO_MEDIO
5,Acessórios,Fone de Ouvido com Fio,4076.0,65,61.050000
7,Acessórios,Mochila Escolar,6203.2,45,141.350000
6,Acessórios,Fone de Ouvido sem Fio,4274.0,40,130.280000
0,Acessórios,Cabo USB,822.8,32,25.733333
9,Acessórios,Suporte de Celular,550.6,28,21.520000
3,Acessórios,Calculadora Simples,668.0,26,27.425000
10,Acessórios,Suporte de Monitor,1353.9,16,87.450000
1,Acessórios,Caixa de Som Bluetooth,2118.6,14,199.900000
4,Acessórios,Carregador Portátil,1645.6,14,122.900000
8,Acessórios,Mochila Executiva,3058.9,12,254.925000


In [15]:
dados_q3[dados_q3["SECAO_PRODUTO"] == "Informática"].head(20)

,SECAO_PRODUTO,CATEGORIA_PRODUTO,FATURAMENTO_BRUTO_TOTAL,QTD_UNIDADES_VENDIDAS_MES_TOTAL,PRECO_UNITARIO_MEDIO
12,Informática,Cartucho de Tinta,11671.0,140,104.900000
18,Informática,Mouse Óptico,3727.2,78,49.900000
17,Informática,Mouse Sem Fio,6117.9,71,89.650000
24,Informática,Teclado Simples,8023.2,68,119.900000
21,Informática,Pen Drive,3693.6,63,61.233333
25,Informática,Webcam,3765.1,49,86.566667
14,Informática,Impressora a Jato,11507.8,22,533.900000
13,Informática,HD Externo,4990.1,19,296.700000
16,Informática,Monitor,16444.8,19,887.150000
22,Informática,Tablet,50775.3,17,2919.700000


In [16]:
dados_q3[dados_q3["SECAO_PRODUTO"] == "Papelaria"].head(30)

,SECAO_PRODUTO,CATEGORIA_PRODUTO,FATURAMENTO_BRUTO_TOTAL,QTD_UNIDADES_VENDIDAS_MES_TOTAL,PRECO_UNITARIO_MEDIO
30,Papelaria,Brinde,3.41,341,0.010000
34,Papelaria,Caneta Simples,722.40,337,2.226316
44,Papelaria,Papéis e Cartolina,980.90,246,4.385714
36,Papelaria,Envelope,183.30,199,1.055556
39,Papelaria,Lápis,249.40,172,1.533333
40,Papelaria,Lápis de Cor e Giz,5416.10,169,37.025000
45,Papelaria,Pasta,1216.80,146,11.230769
33,Papelaria,Caneta Especial,2696.00,130,34.828571
31,Papelaria,Caderno 1 Matéria,1060.60,117,8.646154
41,Papelaria,Marca Texto,718.20,108,6.988889


Insira a descrição dos resultados / conclusões aqui
Na seção de Papelaria, podemos verificar que as 5 categorias mais vendidas possuem valor unitário menor que R$5,00. A partir da sexta categoria mais vendida, os valores unitários variam e, não necessariamente os produtos mais baratos vendem mais.
Já na seção de Informática vemos o oposto: as 3 categorias que menos vendem possuem os maiores valores unitários. Vemos também que os produtos que mais vendem são mais baratos, porém novamente, não necessariamente os produtos mais baratos são os mais vendidos.
A secão de Acessórios, talvez seja a que existe maior variação. Vemos que os produtos mais baratos estão no meio do ranking dos mais vendidos (Cabo USB, suporte de celular, calculadora). No entanto, vemos que o produto mais vendido e o menos vendido possuem valores unitários "medianos", ou seja, não pertencem ao grupo dos produtos mais baratos, nem dos mais caros.

### **QUESTÃO 4**

#### Considerando os produtos com maior faturamento bruto, quantos deles foram responsáveis por 80% do faturamento bruto total da loja?

In [17]:
# Insira seus códigos em Python aqui (divida em várias células, se desejar)
# Considerando os produtos com maior faturamento, vamos obter a quantidade que representa os 20% menores faturamentos.
x = dados['FATURAMENTO_BRUTO'].quantile(0.2)
x

np.float64(32.220000000000006)

In [18]:
# Os 20% produtos de menor faturamento tiveram faturamento menos ou igual a R$32,22. Vamos descobrir quantos produtos estão no mesmo ou abaixo desse valor.
menor_20 = dados[dados['FATURAMENTO_BRUTO']  <= x].shape[0]

menor_20

58

In [19]:
# Total de produtos existentes na base
total = dados.shape[0]

total

290

In [20]:
prod_80 = total - menor_20

prod_80

232

Insira a descrição dos resultados / conclusões aqui
Considerando os produtos de maior faturamento bruto, verificamos que 232 produtos representam 80% do faturamento bruto da loja. Se tivéssemos calculado o percentil 80 do faturamento bruto, teríamos obtido o mesmo resultado.

### **QUESTÃO 5**

#### Quais são os 3 produtos com menor faturamento bruto em cada seção?

In [21]:
dados.head(20)

,COD_PRODUTO,CATEGORIA_PRODUTO,SECAO_PRODUTO,PRECO_UNITARIO,QTD_UNIDADES_VENDIDAS_MES,FATURAMENTO_BRUTO
0,COD_P_233209,Tinta e Pincel,Papelaria,5.9,14,82.6
1,COD_P_235103,Caneta Especial,Papelaria,5.9,26,153.4
2,COD_P_178413,Monitor,Informática,799.9,3,2399.7
3,COD_P_316368,Mochila Escolar,Acessórios,139.0,3,417.0
4,COD_P_190632,Suporte de Celular,Acessórios,29.9,5,149.5
5,COD_P_168517,Tinta e Pincel,Papelaria,7.9,6,47.4
6,COD_P_220491,Papéis e Cartolina,Papelaria,14.9,6,89.4
7,COD_P_310383,Impressora a Jato,Informática,399.9,7,2799.3
8,COD_P_231329,Cartucho de Tinta,Informática,139.9,12,1678.8
9,COD_P_309533,Grampeador e Perfurador,Papelaria,28.9,2,57.8


In [25]:
# Insira seus códigos em Python aqui (divida em várias células, se desejar)
dados_q5 = dados

dados_q5["rank"] = dados_q5.groupby("SECAO_PRODUTO")["FATURAMENTO_BRUTO"].rank(method="first", ascending=True)

dados_q5.sort_values("rank", ascending=True).head(100)

,COD_PRODUTO,CATEGORIA_PRODUTO,SECAO_PRODUTO,PRECO_UNITARIO,QTD_UNIDADES_VENDIDAS_MES,FATURAMENTO_BRUTO,rank
69,COD_P_324413,Impressora a Jato,Informática,629.9,0,0.0,1.0
53,COD_P_316501,Agenda,Papelaria,65.9,0,0.0,1.0
112,COD_P_321815,Calculadora Científica,Acessórios,89.9,0,0.0,1.0
98,COD_P_211988,Notebook,Informática,4399.9,0,0.0,2.0
126,COD_P_315008,Tripé,Acessórios,47.9,0,0.0,2.0
...,...,...,...,...,...,...,...
55,COD_P_258443,Cabo USB,Acessórios,23.9,12,286.8,32.0
83,COD_P_223412,Caixa de Som Bluetooth,Acessórios,294.9,1,294.9,33.0
216,COD_P_221674,Caneta Simples,Papelaria,1.5,13,19.5,33.0
270,COD_P_222924,Cartucho de Tinta,Informática,54.9,40,2196.0,33.0


In [29]:
dados_q5[dados_q5["rank"] <= 3].sort_values(["SECAO_PRODUTO", "rank"], ascending=[True, True])

,COD_PRODUTO,CATEGORIA_PRODUTO,SECAO_PRODUTO,PRECO_UNITARIO,QTD_UNIDADES_VENDIDAS_MES,FATURAMENTO_BRUTO,rank
112,COD_P_321815,Calculadora Científica,Acessórios,89.9,0,0.0,1.0
126,COD_P_315008,Tripé,Acessórios,47.9,0,0.0,2.0
153,COD_P_244216,Fone de Ouvido sem Fio,Acessórios,139.8,0,0.0,3.0
69,COD_P_324413,Impressora a Jato,Informática,629.9,0,0.0,1.0
98,COD_P_211988,Notebook,Informática,4399.9,0,0.0,2.0
144,COD_P_316388,Notebook Premium,Informática,4999.9,0,0.0,3.0
53,COD_P_316501,Agenda,Papelaria,65.9,0,0.0,1.0
124,COD_P_303835,Estilete,Papelaria,8.9,0,0.0,2.0
162,COD_P_136800,Marca Texto,Papelaria,6.9,0,0.0,3.0


Insira a descrição dos resultados / conclusões aqui
O enunciado não especifica se devemos considerar TODOS os produtos, incluindo os que não faturaram, ou apenas os que tiveram algum faturamento. O enunciado também se refere a PRODUTOS e não a categoria ou seção. Portanto, realizei a análise considerando os códigos do produto.
Nesse caso, vemos que, para a secão Acessórios, os 3 produtos que menos faturaram foram:
    1) Calculadora Científica - COD_P_321815
    2) Tripé - COD_P_315008
    3) Fone de Ouvido sem fio - COD_P_244216

Para a seção Informática, os 3 produtos que menos faturaram foram:
    1) COD_P_324413	Impressora a Jato
    2) COD_P_211988	Notebook
    3) COD_P_316388	Notebook Premium

Para a seção de Papelaria, os 3 produtos que menos faturaram foram:
    1) COD_P_316501	Agenda
    2) COD_P_303835	Estilete
    3) COD_P_136800	Marca Texto

### **QUESTÃO 6**

#### Existem produtos com valores atípicos de quantidade de unidades vendidas? E de preço unitário? E de faturamento bruto?

In [32]:
# Insira seus códigos em Python aqui (divida em várias células, se desejar)
# Calculando o Percentil 1 e o Percentil 99 de unidades vendidas:
percentil_1_qt = dados['QTD_UNIDADES_VENDIDAS_MES'].quantile(0.01)
percentil_99_qt = dados['QTD_UNIDADES_VENDIDAS_MES'].quantile(0.99)

# Calculando o Percentil 1 e o Percentil 99 do Preço Unitário:
percentil_1_preco = dados['PRECO_UNITARIO'].quantile(0.01)
percentil_99_preco = dados['PRECO_UNITARIO'].quantile(0.99)

# Calculando o Percentil 1 e o Percentil 99 do faturamento bruto:
percentil_1_fat = dados['FATURAMENTO_BRUTO'].quantile(0.01)
percentil_99_fat = dados['FATURAMENTO_BRUTO'].quantile(0.99)

display(percentil_1_qt, percentil_99_qt, percentil_1_preco, percentil_99_preco,
        percentil_1_fat, percentil_99_fat)

np.float64(0.0)

np.float64(74.55000000000007)

np.float64(0.01)

np.float64(4443.9110000000055)

np.float64(0.0)

np.float64(6929.382000000041)

In [38]:
dados[dados["QTD_UNIDADES_VENDIDAS_MES"] <= percentil_1_qt].head(20)

,COD_PRODUTO,CATEGORIA_PRODUTO,SECAO_PRODUTO,PRECO_UNITARIO,QTD_UNIDADES_VENDIDAS_MES,FATURAMENTO_BRUTO,rank
53,COD_P_316501,Agenda,Papelaria,65.9,0,0.0,1.0
69,COD_P_324413,Impressora a Jato,Informática,629.9,0,0.0,1.0
98,COD_P_211988,Notebook,Informática,4399.9,0,0.0,2.0
112,COD_P_321815,Calculadora Científica,Acessórios,89.9,0,0.0,1.0
124,COD_P_303835,Estilete,Papelaria,8.9,0,0.0,2.0
126,COD_P_315008,Tripé,Acessórios,47.9,0,0.0,2.0
144,COD_P_316388,Notebook Premium,Informática,4999.9,0,0.0,3.0
153,COD_P_244216,Fone de Ouvido sem Fio,Acessórios,139.8,0,0.0,3.0
162,COD_P_136800,Marca Texto,Papelaria,6.9,0,0.0,3.0
194,COD_P_173986,Tripé,Acessórios,63.0,0,0.0,4.0


In [35]:
dados[dados["QTD_UNIDADES_VENDIDAS_MES"] >= percentil_99_qt].head(10)

,COD_PRODUTO,CATEGORIA_PRODUTO,SECAO_PRODUTO,PRECO_UNITARIO,QTD_UNIDADES_VENDIDAS_MES,FATURAMENTO_BRUTO,rank
80,COD_B_000003,Brinde,Papelaria,0.01,132,1.32,10.0
164,COD_P_291073,Envelope,Papelaria,0.80,79,63.20,92.0
268,COD_B_000004,Brinde,Papelaria,0.01,98,0.98,8.0


In [41]:
dados[dados["PRECO_UNITARIO"] <= percentil_1_preco].head(10)

,COD_PRODUTO,CATEGORIA_PRODUTO,SECAO_PRODUTO,PRECO_UNITARIO,QTD_UNIDADES_VENDIDAS_MES,FATURAMENTO_BRUTO,rank
80,COD_B_000003,Brinde,Papelaria,0.01,132,1.32,10.0
147,COD_B_000001,Brinde,Papelaria,0.01,62,0.62,7.0
186,COD_B_000002,Brinde,Papelaria,0.01,49,0.49,6.0
268,COD_B_000004,Brinde,Papelaria,0.01,98,0.98,8.0


In [42]:
dados[dados["PRECO_UNITARIO"] >= percentil_99_preco].head(20)


,COD_PRODUTO,CATEGORIA_PRODUTO,SECAO_PRODUTO,PRECO_UNITARIO,QTD_UNIDADES_VENDIDAS_MES,FATURAMENTO_BRUTO,rank
144,COD_P_316388,Notebook Premium,Informática,4999.9,0,0.0,3.0
179,COD_P_146931,Notebook Premium,Informática,4800.0,1,4800.0,48.0
241,COD_P_255913,Notebook Premium,Informática,5850.0,1,5850.0,51.0


In [43]:
dados[dados["FATURAMENTO_BRUTO"] <= percentil_1_fat].head(10)

,COD_PRODUTO,CATEGORIA_PRODUTO,SECAO_PRODUTO,PRECO_UNITARIO,QTD_UNIDADES_VENDIDAS_MES,FATURAMENTO_BRUTO,rank
53,COD_P_316501,Agenda,Papelaria,65.9,0,0.0,1.0
69,COD_P_324413,Impressora a Jato,Informática,629.9,0,0.0,1.0
98,COD_P_211988,Notebook,Informática,4399.9,0,0.0,2.0
112,COD_P_321815,Calculadora Científica,Acessórios,89.9,0,0.0,1.0
124,COD_P_303835,Estilete,Papelaria,8.9,0,0.0,2.0
126,COD_P_315008,Tripé,Acessórios,47.9,0,0.0,2.0
144,COD_P_316388,Notebook Premium,Informática,4999.9,0,0.0,3.0
153,COD_P_244216,Fone de Ouvido sem Fio,Acessórios,139.8,0,0.0,3.0
162,COD_P_136800,Marca Texto,Papelaria,6.9,0,0.0,3.0
194,COD_P_173986,Tripé,Acessórios,63.0,0,0.0,4.0


In [44]:
dados[dados["FATURAMENTO_BRUTO"] >= percentil_99_fat].head(10)

,COD_PRODUTO,CATEGORIA_PRODUTO,SECAO_PRODUTO,PRECO_UNITARIO,QTD_UNIDADES_VENDIDAS_MES,FATURAMENTO_BRUTO,rank
54,COD_P_263117,Tablet,Informática,2980.0,6,17880.0,57.0
66,COD_P_344598,Tablet,Informática,2399.0,4,9596.0,55.0
278,COD_P_279132,Tablet,Informática,4399.9,4,17599.6,56.0


Insira a descrição dos resultados / conclusões aqui:
Novamente, o enunciado fala de PRODUTOS e não de categorias ou seção.
Para a quantidade de unidades vendidas no mês, não vemos valores atípicos para baixo, uma vez que o percentil 1 para esta variável já é ZERO. Para cima, encontramos 3 produtos com quantidades vendidas maior que o P99, 74,55 unidades: 
    1) COD_B_000003	Brinde
    2) COD_P_291073	Envelope
    3) COD_B_000004	Brinde

Com relação ao preço unitário, não encontramos nenhum valor atípico para baixo, já que o valor de P01 já é de R$0,01. Para cima, encontramos 3 produtos com preço unitário acima de R$4443,91:
    1) COD_P_316388	Notebook Premium
    2) COD_P_146931	Notebook Premium
    3) COD_P_255913	Notebook Premium

Para o faturamento bruto, novamente vemos que não há valores atípicos para baixo, pois o P01 para esta variável já é zero. Para cima, encontramos 3 produtos com faturamento bruto maior que R$ 69929,38. São eles:
    1) COD_P_263117	Tablet	
    2) COD_P_344598	Tablet
    3) COD_P_279132	Tablet	

### **QUESTÃO 7**

#### Qual seção de produtos tem maior variabilidade de preços unitários?

In [53]:
# Insira seus códigos em Python aqui (divida em várias células, se desejar)
dados_q7 = dados.groupby("SECAO_PRODUTO")["PRECO_UNITARIO"].agg(["min", "max"])

In [54]:
dados_q7["diferença"] = dados_q7["max"] - dados_q7["min"]

dados_q7

,min,max,diferença
SECAO_PRODUTO,,,
Acessórios,12.90,294.9,282.00
Informática,31.90,5850.0,5818.10
Papelaria,0.01,129.9,129.89


Insira a descrição dos resultados / conclusões aqui
A seção com maior variabilidade de preços unitários é a seção de Informática , com preços que variam de R$31,90 a R$5850,00.

### **QUESTÃO 8**

#### Existem produtos com alta quantidade de unidades vendidas e baixo preço? Existem produtos com baixa quantidade de unidades vendidas e alto preço?

In [55]:
# Insira seus códigos em Python aqui (divida em várias células, se desejar)
dados[dados["QTD_UNIDADES_VENDIDAS_MES"] >= percentil_99_qt]

,COD_PRODUTO,CATEGORIA_PRODUTO,SECAO_PRODUTO,PRECO_UNITARIO,QTD_UNIDADES_VENDIDAS_MES,FATURAMENTO_BRUTO,rank
80,COD_B_000003,Brinde,Papelaria,0.01,132,1.32,10.0
164,COD_P_291073,Envelope,Papelaria,0.80,79,63.20,92.0
268,COD_B_000004,Brinde,Papelaria,0.01,98,0.98,8.0


In [57]:
dados[dados["QTD_UNIDADES_VENDIDAS_MES"] <= percentil_1_qt].sort_values("PRECO_UNITARIO", ascending=False).head(10)

,COD_PRODUTO,CATEGORIA_PRODUTO,SECAO_PRODUTO,PRECO_UNITARIO,QTD_UNIDADES_VENDIDAS_MES,FATURAMENTO_BRUTO,rank
144,COD_P_316388,Notebook Premium,Informática,4999.9,0,0.0,3.0
98,COD_P_211988,Notebook,Informática,4399.9,0,0.0,2.0
69,COD_P_324413,Impressora a Jato,Informática,629.9,0,0.0,1.0
153,COD_P_244216,Fone de Ouvido sem Fio,Acessórios,139.8,0,0.0,3.0
112,COD_P_321815,Calculadora Científica,Acessórios,89.9,0,0.0,1.0
53,COD_P_316501,Agenda,Papelaria,65.9,0,0.0,1.0
194,COD_P_173986,Tripé,Acessórios,63.0,0,0.0,4.0
126,COD_P_315008,Tripé,Acessórios,47.9,0,0.0,2.0
124,COD_P_303835,Estilete,Papelaria,8.9,0,0.0,2.0
282,COD_P_318311,Tinta e Pincel,Papelaria,8.9,0,0.0,5.0


Insira a descrição dos resultados / conclusões aqui
Podemos que ver que os valores atípicos para cima de quantidade de unidades vendidas são de produtos com baixo valor unitário. Todos os 3 produtos tem valor unitário abaixo de R$1,00.
Quanto aos produtos com alto valor unitário temos 2 produtos da seção de informatica encabeçando a lista, e que custam mais de R$4.000,00 . os dois tiveram zero unidades vendidas no mês.